### ISSUE TITLE
linter: `jsx-a11y/img-redundant-alt`: False detection when "image" is part of a larger word in alt tex

### ISSUE THREAD

**Issue by akahoshi1421** [oxc-project/oxc#19336]
### What version of Oxlint are you using?

1.47.0

### What command did you run?

npx oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [],
  "categories": {
    "correctness": "off"
  },
  "env": {
    "builtin": true
  },
  "ignorePatterns": [
    "dist"
  ],
  "overrides": [
    {
      "files": [
        "src/*.{ts,tsx}"
      ],
      "rules": {
        "react/no-array-index-key": "error",
        "jsx-a11y/img-redundant-alt": "error"
      },
      "globals": {
        "AudioWorkletGlobalScope": "readonly",
        "AudioWorkletProcessor": "readonly",
        "currentFrame": "readonly",
        "currentTime": "readonly",
        "registerProcessor": "readonly",
        "sampleRate": "readonly",
        "WorkletGlobalScope": "readonly"
      },
      "env": {
        "es2020": true,
        "browser": true
      },
      "plugins": [
        "react",
        "jsx-a11y"
      ]
    }
  ]
}
```

### What happened?

ESLint's `eslint-plugin-jsx-a11y` implementation correctly uses whole-word matching, but oxlint appears to use substring matching, causing false positives.

Example
```tsx
import "./App.css";

export function App() {

  return (
    <>
      {/* jsx-a11y/img-redundant-alt is deteced (normal detection) */}
      <img src="https://dummyimage.com/600x400/000/fff" alt="hello Image" />

      {/* jsx-a11y/img-redundant-alt is deteced (false detection) */}
      <img src="https://dummyimage.com/600x400/000/fff" alt="helloImage" />
    </>
  );
}
```

Comparison with ESLint
<img width="879" height="393" alt="Image" src="https://github.com/user-attachments/assets/d150c5a0-be3a-41bb-ac34-2c3b941bbc64" />

**Comment by TheAlexLichter** [comment]
[REPL link](https://playground.oxc.rs/?code=%0Aexport+function+App%28%29+%7B%0A%0A++return+%28%0A++++%3C%3E%0A++++++%7B%2F*+jsx-a11y%2Fimg-redundant-alt+is+deteced+%28normal+detection%29+*%2F%7D%0A++++++%3Cimg+src%3D%22https%3A%2F%2Fdummyimage.com%2F600x400%2F000%2Ffff%22+alt%3D%22hello+Image%22+%2F%3E%0A%0A++++++%7B%2F*+jsx-a11y%2Fimg-redundant-alt+is+deteced+%28false+detection%29+*%2F%7D%0A++++++%3Cimg+src%3D%22https%3A%2F%2Fdummyimage.com%2F600x400%2F000%2Ffff%22+alt%3D%22helloImage%22+%2F%3E%0A++++%3C%2F%3E%0A++%29%3B%0A%7D)
